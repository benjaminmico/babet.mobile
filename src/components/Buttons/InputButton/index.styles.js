import styled from 'styled-components'

export const InputButtonContainer = styled.TouchableOpacity`
  background-color: ${props => props.backgroundColor};
  width: 339px;
  height: 53px;
  padding-top: 6px;
  padding-left: 9px;
  padding-right: 9px;
  padding-bottom: 6px;
  justify-content: center;
  align-items: center;
`

export const InputButtonText = styled.Text`
  color: ${props => props.color};
  font-family: 'AvenirNext-Bold';
  font-size: 14px;
  line-height: 19px;
  letter-spacing: 1px;
  text-align: center;
`
