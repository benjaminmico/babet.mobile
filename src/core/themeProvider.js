import AsyncStorage from '@react-native-community/async-storage'
import React, {useContext, useEffect, useState} from 'react'
import {useDarkMode} from 'react-native-dark-mode'
import THEMES from './themes.json'

const STORAGE_KEY = 'THEME_ID'
const ThemeContext = React.createContext()

export const ThemeContextProvider = ({children}) => {
  const [themeID, setThemeID] = useState()

  const isDarkMode = useDarkMode ? useDarkMode() : false

  useEffect(() => {
    ;(async () => {
      // eslint-disable-next-line no-unused-expressions
      isDarkMode ? setThemeID(THEMES[0].key) : setThemeID(THEMES[1].key)
      // const storedThemeID = await AsyncStorage.getItem(STORAGE_KEY)
      // if (storedThemeID) setThemeID(storedThemeID)
      // // eslint-disable-next-line no-unused-expressions
      // else isDarkMode ? setThemeID(THEMES[0].key) : setThemeID(THEMES[1].key)
    })()
  }, [])

  return <ThemeContext.Provider value={{themeID, setThemeID}}>{themeID ? children : null}</ThemeContext.Provider>
}

export function withTheme(Component) {
  return props => {
    const {themeID, setThemeID} = useContext(ThemeContext)

    const getTheme = themeID => THEMES.find(theme => theme.key === themeID)
    const setTheme = themeID => {
      AsyncStorage.setItem(STORAGE_KEY, themeID)
      setThemeID(themeID)
    }

    return <Component {...props} themes={THEMES} theme={getTheme(themeID)} setTheme={setTheme} />
  }
}

export function getTheme() {
  const {themeID} = useContext(ThemeContext)

  const getThemeContent = themeID => THEMES.find(theme => theme.key === themeID)

  return getThemeContent(themeID)
}
