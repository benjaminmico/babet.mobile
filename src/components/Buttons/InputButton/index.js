// @flow

import {withTheme} from '@core/themeProvider'
import React from 'react'
import {InputButtonContainer, InputButtonText} from './index.styles'

type Props = {
  // input button label
  label: String,
  // on press input button
  onPress: () => void,
  // theme
  theme: Object,
}

const InputButton = ({label, onPress, theme}: Props) => {
  // get theme props
  const {
    colors: {
      components: {backgroundInputButton},
      texts: {
        buttons: {isPrimaryAction: isPrimaryActionInputButtonTextColor},
      },
    },
  } = theme

  /**
   * render input button
   * text can only be de displayed on one line
   * */

  return (
    <InputButtonContainer disabled={!onPress} onPress={onPress} backgroundColor={backgroundInputButton}>
      <InputButtonText numberOfLines={1} color={isPrimaryActionInputButtonTextColor}>
        {label.toUpperCase()}
      </InputButtonText>
    </InputButtonContainer>
  )
}

export default withTheme(InputButton)
