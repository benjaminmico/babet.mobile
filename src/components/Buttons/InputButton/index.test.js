/**
 * @jest-environment jsdom
 */
import React from 'react'
import {StyleSheet} from 'react-native'
import {ThemeContextProvider} from '@core/themeProvider'
import {mount} from 'enzyme'
import InputButton from '.'

describe.only('InputButton', () => {
  describe('Integrity tests', () => {
    it('renders InputButton correctly', () => {
      const label = 'Supprimer'

      const wrapper = mount(
        <ThemeContextProvider>
          <InputButton label={label} />
        </ThemeContextProvider>,
      )

      const InputButtonText = wrapper.find('Text').first()
      expect(InputButtonText.exists()).toBe(true)
      expect(InputButtonText.prop('children')).toBe(label.toUpperCase())
    })
  })

  describe('Props tests', () => {
    it('renders default InputButton', () => {
      const label = 'Supprimer'

      const wrapper = mount(
        <ThemeContextProvider>
          <InputButton label={label} />
        </ThemeContextProvider>,
      )
      const InputButtonContainer = wrapper.find('TouchableOpacity').first()
      const InputButtonTextContainer = wrapper.find('Text').first()

      expect(wrapper).toMatchSnapshot()

      const stylesContainer = StyleSheet.flatten(InputButtonContainer.prop('style'))
      expect(stylesContainer.backgroundColor).toBe('#5100FF')
      expect(stylesContainer.width).toBe(339)
      expect(stylesContainer.height).toBe(53)
      expect(stylesContainer.paddingTop).toBe(6)
      expect(stylesContainer.paddingLeft).toBe(9)
      expect(stylesContainer.paddingRight).toBe(9)
      expect(stylesContainer.paddingBottom).toBe(6)
      expect(stylesContainer.justifyContent).toBe('center')
      expect(stylesContainer.alignItems).toBe('center')

      const stylesText = StyleSheet.flatten(InputButtonTextContainer.prop('style'))
      expect(stylesText.color).toBe('#FFFFFF')
      expect(stylesText.fontFamily).toBe('AvenirNext-Bold')
      expect(stylesText.fontSize).toBe(14)
      expect(stylesText.lineHeight).toBe(19)
      expect(stylesText.letterSpacing).toBe(1)
      expect(stylesText.textAlign).toBe('center')
    })
  })
})
