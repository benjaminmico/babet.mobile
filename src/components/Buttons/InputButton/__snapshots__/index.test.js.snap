// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputButton Props tests renders default InputButton 1`] = `
<ThemeContextProvider>
  <Component
    label="Supprimer"
  >
    <InputButton
      label="Supprimer"
      setTheme={[Function]}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "color": "#101112",
          "colors": Object {
            "components": Object {
              "background": "#FFFFFF",
              "backgroundAction": "#F3F2F2",
              "backgroundContainer": "#FFFFFF",
              "backgroundFilter": "#F3F2F2",
              "backgroundInputButton": "#5100FF",
              "backgroundPrimaryAction": "#5100FF",
              "icon": "#000000",
            },
            "palette": Object {
              "black": "#000000",
              "comparisonDescription": "#A3A2A2",
              "error": "#F34949",
              "line": "#D4D8D9",
              "lost": "#CA1010",
              "pending": "#7C7C7C",
              "purple": "#5100FF",
              "separator": "#979797",
              "white": "#FFFFFF",
              "won": "#14B916",
            },
            "shadows": Object {},
            "texts": Object {
              "buttons": Object {
                "actions": "#5100FF",
                "cancellable": "#3E3E3E",
                "filter": "#3E3E3E",
                "isPrimaryAction": "#FFFFFF",
                "social": Object {
                  "apple": "#000000",
                  "facebook": "#1976D2",
                  "gmail": "#F44336",
                },
              },
              "description": "#ADADAD",
              "filters": Object {
                "multiFilter": "#5100FF",
                "selected": "#000000",
                "unselected": "#A3A2A2",
              },
              "input": Object {
                "label": "#7C7C7C",
                "value": "#000000",
              },
              "labelInput": "#7C7C7C",
              "reverseText": "#FFFFFF",
              "stats": Object {
                "kpiValue": "#3E3E3E",
              },
              "text": "#000000",
              "title": "#000000",
            },
          },
          "isShadow": true,
          "key": "white",
          "statusBar": "dark-content",
        }
      }
      themes={
        Array [
          Object {
            "backgroundColor": "#101112",
            "color": "#ffffff",
            "colors": Object {
              "components": Object {
                "background": "#101112",
                "backgroundAction": "#F3F2F2",
                "backgroundContainer": "#2F3134",
                "backgroundFilter": "#F3F2F2",
                "backgroundInputButton": "#5100FF",
                "backgroundPrimaryAction": "#5100FF",
                "icon": "#FFFFFF",
              },
              "palette": Object {
                "black": "#000000",
                "comparisonDescription": "#A3A2A2",
                "error": "#F34949",
                "line": "#D4D8D9",
                "lost": "#CA1010",
                "pending": "#7C7C7C",
                "purple": "#5100FF",
                "separator": "#979797",
                "white": "#FFFFFF",
                "won": "#14B916",
              },
              "texts": Object {
                "buttons": Object {
                  "actions": "#5100FF",
                  "cancellable": "#3E3E3E",
                  "filter": "#3E3E3E",
                  "isPrimaryAction": "#FFFFFF",
                  "social": Object {
                    "apple": "#000000",
                    "facebook": "#1976D2",
                    "gmail": "#F44336",
                  },
                },
                "description": "#ADADAD",
                "filters": Object {
                  "multiFilter": "#5100FF",
                  "selected": "#000000",
                  "unselected": "#A3A2A2",
                },
                "input": Object {
                  "label": "#7C7C7C",
                  "value": "#FFFFFF",
                },
                "reverseText": "#FFFFFF",
                "stats": Object {
                  "kpiValue": "#FFFFFF",
                },
                "text": "#FFFFFF",
                "title": "#FFFFFF",
              },
            },
            "isShadow": false,
            "key": "dark",
            "statusBar": "light-content",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#101112",
            "colors": Object {
              "components": Object {
                "background": "#FFFFFF",
                "backgroundAction": "#F3F2F2",
                "backgroundContainer": "#FFFFFF",
                "backgroundFilter": "#F3F2F2",
                "backgroundInputButton": "#5100FF",
                "backgroundPrimaryAction": "#5100FF",
                "icon": "#000000",
              },
              "palette": Object {
                "black": "#000000",
                "comparisonDescription": "#A3A2A2",
                "error": "#F34949",
                "line": "#D4D8D9",
                "lost": "#CA1010",
                "pending": "#7C7C7C",
                "purple": "#5100FF",
                "separator": "#979797",
                "white": "#FFFFFF",
                "won": "#14B916",
              },
              "shadows": Object {},
              "texts": Object {
                "buttons": Object {
                  "actions": "#5100FF",
                  "cancellable": "#3E3E3E",
                  "filter": "#3E3E3E",
                  "isPrimaryAction": "#FFFFFF",
                  "social": Object {
                    "apple": "#000000",
                    "facebook": "#1976D2",
                    "gmail": "#F44336",
                  },
                },
                "description": "#ADADAD",
                "filters": Object {
                  "multiFilter": "#5100FF",
                  "selected": "#000000",
                  "unselected": "#A3A2A2",
                },
                "input": Object {
                  "label": "#7C7C7C",
                  "value": "#000000",
                },
                "labelInput": "#7C7C7C",
                "reverseText": "#FFFFFF",
                "stats": Object {
                  "kpiValue": "#3E3E3E",
                },
                "text": "#000000",
                "title": "#000000",
              },
            },
            "isShadow": true,
            "key": "white",
            "statusBar": "dark-content",
          },
        ]
      }
    >
      <Styled(Component)
        backgroundColor="#5100FF"
        disabled={true}
        setTheme={[Function]}
        themes={
          Array [
            Object {
              "backgroundColor": "#101112",
              "color": "#ffffff",
              "colors": Object {
                "components": Object {
                  "background": "#101112",
                  "backgroundAction": "#F3F2F2",
                  "backgroundContainer": "#2F3134",
                  "backgroundFilter": "#F3F2F2",
                  "backgroundInputButton": "#5100FF",
                  "backgroundPrimaryAction": "#5100FF",
                  "icon": "#FFFFFF",
                },
                "palette": Object {
                  "black": "#000000",
                  "comparisonDescription": "#A3A2A2",
                  "error": "#F34949",
                  "line": "#D4D8D9",
                  "lost": "#CA1010",
                  "pending": "#7C7C7C",
                  "purple": "#5100FF",
                  "separator": "#979797",
                  "white": "#FFFFFF",
                  "won": "#14B916",
                },
                "texts": Object {
                  "buttons": Object {
                    "actions": "#5100FF",
                    "cancellable": "#3E3E3E",
                    "filter": "#3E3E3E",
                    "isPrimaryAction": "#FFFFFF",
                    "social": Object {
                      "apple": "#000000",
                      "facebook": "#1976D2",
                      "gmail": "#F44336",
                    },
                  },
                  "description": "#ADADAD",
                  "filters": Object {
                    "multiFilter": "#5100FF",
                    "selected": "#000000",
                    "unselected": "#A3A2A2",
                  },
                  "input": Object {
                    "label": "#7C7C7C",
                    "value": "#FFFFFF",
                  },
                  "reverseText": "#FFFFFF",
                  "stats": Object {
                    "kpiValue": "#FFFFFF",
                  },
                  "text": "#FFFFFF",
                  "title": "#FFFFFF",
                },
              },
              "isShadow": false,
              "key": "dark",
              "statusBar": "light-content",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#101112",
              "colors": Object {
                "components": Object {
                  "background": "#FFFFFF",
                  "backgroundAction": "#F3F2F2",
                  "backgroundContainer": "#FFFFFF",
                  "backgroundFilter": "#F3F2F2",
                  "backgroundInputButton": "#5100FF",
                  "backgroundPrimaryAction": "#5100FF",
                  "icon": "#000000",
                },
                "palette": Object {
                  "black": "#000000",
                  "comparisonDescription": "#A3A2A2",
                  "error": "#F34949",
                  "line": "#D4D8D9",
                  "lost": "#CA1010",
                  "pending": "#7C7C7C",
                  "purple": "#5100FF",
                  "separator": "#979797",
                  "white": "#FFFFFF",
                  "won": "#14B916",
                },
                "shadows": Object {},
                "texts": Object {
                  "buttons": Object {
                    "actions": "#5100FF",
                    "cancellable": "#3E3E3E",
                    "filter": "#3E3E3E",
                    "isPrimaryAction": "#FFFFFF",
                    "social": Object {
                      "apple": "#000000",
                      "facebook": "#1976D2",
                      "gmail": "#F44336",
                    },
                  },
                  "description": "#ADADAD",
                  "filters": Object {
                    "multiFilter": "#5100FF",
                    "selected": "#000000",
                    "unselected": "#A3A2A2",
                  },
                  "input": Object {
                    "label": "#7C7C7C",
                    "value": "#000000",
                  },
                  "labelInput": "#7C7C7C",
                  "reverseText": "#FFFFFF",
                  "stats": Object {
                    "kpiValue": "#3E3E3E",
                  },
                  "text": "#000000",
                  "title": "#000000",
                },
              },
              "isShadow": true,
              "key": "white",
              "statusBar": "dark-content",
            },
          ]
        }
      >
        <StyledNativeComponent
          backgroundColor="#5100FF"
          disabled={true}
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "displayName": "Styled(Component)",
              "inlineStyle": InlineStyle {
                "rules": Array [
                  "
  background-color: ",
                  [Function],
                  ";
  width: 100%;
  height: 39px;
  padding-top: 6px;
  padding-left: 9px;
  padding-right: 9px;
  padding-bottom: 6px;
  justify-content: center;
  align-items: center;
",
                ],
              },
              "render": [Function],
              "styledComponentId": "StyledNativeComponent",
              "target": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          setTheme={[Function]}
          themes={
            Array [
              Object {
                "backgroundColor": "#101112",
                "color": "#ffffff",
                "colors": Object {
                  "components": Object {
                    "background": "#101112",
                    "backgroundAction": "#F3F2F2",
                    "backgroundContainer": "#2F3134",
                    "backgroundFilter": "#F3F2F2",
                    "backgroundInputButton": "#5100FF",
                    "backgroundPrimaryAction": "#5100FF",
                    "icon": "#FFFFFF",
                  },
                  "palette": Object {
                    "black": "#000000",
                    "comparisonDescription": "#A3A2A2",
                    "error": "#F34949",
                    "line": "#D4D8D9",
                    "lost": "#CA1010",
                    "pending": "#7C7C7C",
                    "purple": "#5100FF",
                    "separator": "#979797",
                    "white": "#FFFFFF",
                    "won": "#14B916",
                  },
                  "texts": Object {
                    "buttons": Object {
                      "actions": "#5100FF",
                      "cancellable": "#3E3E3E",
                      "filter": "#3E3E3E",
                      "isPrimaryAction": "#FFFFFF",
                      "social": Object {
                        "apple": "#000000",
                        "facebook": "#1976D2",
                        "gmail": "#F44336",
                      },
                    },
                    "description": "#ADADAD",
                    "filters": Object {
                      "multiFilter": "#5100FF",
                      "selected": "#000000",
                      "unselected": "#A3A2A2",
                    },
                    "input": Object {
                      "label": "#7C7C7C",
                      "value": "#FFFFFF",
                    },
                    "reverseText": "#FFFFFF",
                    "stats": Object {
                      "kpiValue": "#FFFFFF",
                    },
                    "text": "#FFFFFF",
                    "title": "#FFFFFF",
                  },
                },
                "isShadow": false,
                "key": "dark",
                "statusBar": "light-content",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#101112",
                "colors": Object {
                  "components": Object {
                    "background": "#FFFFFF",
                    "backgroundAction": "#F3F2F2",
                    "backgroundContainer": "#FFFFFF",
                    "backgroundFilter": "#F3F2F2",
                    "backgroundInputButton": "#5100FF",
                    "backgroundPrimaryAction": "#5100FF",
                    "icon": "#000000",
                  },
                  "palette": Object {
                    "black": "#000000",
                    "comparisonDescription": "#A3A2A2",
                    "error": "#F34949",
                    "line": "#D4D8D9",
                    "lost": "#CA1010",
                    "pending": "#7C7C7C",
                    "purple": "#5100FF",
                    "separator": "#979797",
                    "white": "#FFFFFF",
                    "won": "#14B916",
                  },
                  "shadows": Object {},
                  "texts": Object {
                    "buttons": Object {
                      "actions": "#5100FF",
                      "cancellable": "#3E3E3E",
                      "filter": "#3E3E3E",
                      "isPrimaryAction": "#FFFFFF",
                      "social": Object {
                        "apple": "#000000",
                        "facebook": "#1976D2",
                        "gmail": "#F44336",
                      },
                    },
                    "description": "#ADADAD",
                    "filters": Object {
                      "multiFilter": "#5100FF",
                      "selected": "#000000",
                      "unselected": "#A3A2A2",
                    },
                    "input": Object {
                      "label": "#7C7C7C",
                      "value": "#000000",
                    },
                    "labelInput": "#7C7C7C",
                    "reverseText": "#FFFFFF",
                    "stats": Object {
                      "kpiValue": "#3E3E3E",
                    },
                    "text": "#000000",
                    "title": "#000000",
                  },
                },
                "isShadow": true,
                "key": "white",
                "statusBar": "dark-content",
              },
            ]
          }
        >
          <ForwardRef
            backgroundColor="#5100FF"
            disabled={true}
            setTheme={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#5100FF",
                  "height": 39,
                  "justifyContent": "center",
                  "paddingBottom": 6,
                  "paddingLeft": 9,
                  "paddingRight": 9,
                  "paddingTop": 6,
                  "width": "100%",
                },
              ]
            }
            themes={
              Array [
                Object {
                  "backgroundColor": "#101112",
                  "color": "#ffffff",
                  "colors": Object {
                    "components": Object {
                      "background": "#101112",
                      "backgroundAction": "#F3F2F2",
                      "backgroundContainer": "#2F3134",
                      "backgroundFilter": "#F3F2F2",
                      "backgroundInputButton": "#5100FF",
                      "backgroundPrimaryAction": "#5100FF",
                      "icon": "#FFFFFF",
                    },
                    "palette": Object {
                      "black": "#000000",
                      "comparisonDescription": "#A3A2A2",
                      "error": "#F34949",
                      "line": "#D4D8D9",
                      "lost": "#CA1010",
                      "pending": "#7C7C7C",
                      "purple": "#5100FF",
                      "separator": "#979797",
                      "white": "#FFFFFF",
                      "won": "#14B916",
                    },
                    "texts": Object {
                      "buttons": Object {
                        "actions": "#5100FF",
                        "cancellable": "#3E3E3E",
                        "filter": "#3E3E3E",
                        "isPrimaryAction": "#FFFFFF",
                        "social": Object {
                          "apple": "#000000",
                          "facebook": "#1976D2",
                          "gmail": "#F44336",
                        },
                      },
                      "description": "#ADADAD",
                      "filters": Object {
                        "multiFilter": "#5100FF",
                        "selected": "#000000",
                        "unselected": "#A3A2A2",
                      },
                      "input": Object {
                        "label": "#7C7C7C",
                        "value": "#FFFFFF",
                      },
                      "reverseText": "#FFFFFF",
                      "stats": Object {
                        "kpiValue": "#FFFFFF",
                      },
                      "text": "#FFFFFF",
                      "title": "#FFFFFF",
                    },
                  },
                  "isShadow": false,
                  "key": "dark",
                  "statusBar": "light-content",
                },
                Object {
                  "backgroundColor": "#ffffff",
                  "color": "#101112",
                  "colors": Object {
                    "components": Object {
                      "background": "#FFFFFF",
                      "backgroundAction": "#F3F2F2",
                      "backgroundContainer": "#FFFFFF",
                      "backgroundFilter": "#F3F2F2",
                      "backgroundInputButton": "#5100FF",
                      "backgroundPrimaryAction": "#5100FF",
                      "icon": "#000000",
                    },
                    "palette": Object {
                      "black": "#000000",
                      "comparisonDescription": "#A3A2A2",
                      "error": "#F34949",
                      "line": "#D4D8D9",
                      "lost": "#CA1010",
                      "pending": "#7C7C7C",
                      "purple": "#5100FF",
                      "separator": "#979797",
                      "white": "#FFFFFF",
                      "won": "#14B916",
                    },
                    "shadows": Object {},
                    "texts": Object {
                      "buttons": Object {
                        "actions": "#5100FF",
                        "cancellable": "#3E3E3E",
                        "filter": "#3E3E3E",
                        "isPrimaryAction": "#FFFFFF",
                        "social": Object {
                          "apple": "#000000",
                          "facebook": "#1976D2",
                          "gmail": "#F44336",
                        },
                      },
                      "description": "#ADADAD",
                      "filters": Object {
                        "multiFilter": "#5100FF",
                        "selected": "#000000",
                        "unselected": "#A3A2A2",
                      },
                      "input": Object {
                        "label": "#7C7C7C",
                        "value": "#000000",
                      },
                      "labelInput": "#7C7C7C",
                      "reverseText": "#FFFFFF",
                      "stats": Object {
                        "kpiValue": "#3E3E3E",
                      },
                      "text": "#000000",
                      "title": "#000000",
                    },
                  },
                  "isShadow": true,
                  "key": "white",
                  "statusBar": "dark-content",
                },
              ]
            }
          >
            <TouchableOpacity
              backgroundColor="#5100FF"
              disabled={true}
              hostRef={null}
              setTheme={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#5100FF",
                    "height": 39,
                    "justifyContent": "center",
                    "paddingBottom": 6,
                    "paddingLeft": 9,
                    "paddingRight": 9,
                    "paddingTop": 6,
                    "width": "100%",
                  },
                ]
              }
              themes={
                Array [
                  Object {
                    "backgroundColor": "#101112",
                    "color": "#ffffff",
                    "colors": Object {
                      "components": Object {
                        "background": "#101112",
                        "backgroundAction": "#F3F2F2",
                        "backgroundContainer": "#2F3134",
                        "backgroundFilter": "#F3F2F2",
                        "backgroundInputButton": "#5100FF",
                        "backgroundPrimaryAction": "#5100FF",
                        "icon": "#FFFFFF",
                      },
                      "palette": Object {
                        "black": "#000000",
                        "comparisonDescription": "#A3A2A2",
                        "error": "#F34949",
                        "line": "#D4D8D9",
                        "lost": "#CA1010",
                        "pending": "#7C7C7C",
                        "purple": "#5100FF",
                        "separator": "#979797",
                        "white": "#FFFFFF",
                        "won": "#14B916",
                      },
                      "texts": Object {
                        "buttons": Object {
                          "actions": "#5100FF",
                          "cancellable": "#3E3E3E",
                          "filter": "#3E3E3E",
                          "isPrimaryAction": "#FFFFFF",
                          "social": Object {
                            "apple": "#000000",
                            "facebook": "#1976D2",
                            "gmail": "#F44336",
                          },
                        },
                        "description": "#ADADAD",
                        "filters": Object {
                          "multiFilter": "#5100FF",
                          "selected": "#000000",
                          "unselected": "#A3A2A2",
                        },
                        "input": Object {
                          "label": "#7C7C7C",
                          "value": "#FFFFFF",
                        },
                        "reverseText": "#FFFFFF",
                        "stats": Object {
                          "kpiValue": "#FFFFFF",
                        },
                        "text": "#FFFFFF",
                        "title": "#FFFFFF",
                      },
                    },
                    "isShadow": false,
                    "key": "dark",
                    "statusBar": "light-content",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                    "color": "#101112",
                    "colors": Object {
                      "components": Object {
                        "background": "#FFFFFF",
                        "backgroundAction": "#F3F2F2",
                        "backgroundContainer": "#FFFFFF",
                        "backgroundFilter": "#F3F2F2",
                        "backgroundInputButton": "#5100FF",
                        "backgroundPrimaryAction": "#5100FF",
                        "icon": "#000000",
                      },
                      "palette": Object {
                        "black": "#000000",
                        "comparisonDescription": "#A3A2A2",
                        "error": "#F34949",
                        "line": "#D4D8D9",
                        "lost": "#CA1010",
                        "pending": "#7C7C7C",
                        "purple": "#5100FF",
                        "separator": "#979797",
                        "white": "#FFFFFF",
                        "won": "#14B916",
                      },
                      "shadows": Object {},
                      "texts": Object {
                        "buttons": Object {
                          "actions": "#5100FF",
                          "cancellable": "#3E3E3E",
                          "filter": "#3E3E3E",
                          "isPrimaryAction": "#FFFFFF",
                          "social": Object {
                            "apple": "#000000",
                            "facebook": "#1976D2",
                            "gmail": "#F44336",
                          },
                        },
                        "description": "#ADADAD",
                        "filters": Object {
                          "multiFilter": "#5100FF",
                          "selected": "#000000",
                          "unselected": "#A3A2A2",
                        },
                        "input": Object {
                          "label": "#7C7C7C",
                          "value": "#000000",
                        },
                        "labelInput": "#7C7C7C",
                        "reverseText": "#FFFFFF",
                        "stats": Object {
                          "kpiValue": "#3E3E3E",
                        },
                        "text": "#000000",
                        "title": "#000000",
                      },
                    },
                    "isShadow": true,
                    "key": "white",
                    "statusBar": "dark-content",
                  },
                ]
              }
            >
              <ForwardRef(AnimatedComponentWrapper)
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#5100FF",
                        "height": 39,
                        "justifyContent": "center",
                        "paddingBottom": 6,
                        "paddingLeft": 9,
                        "paddingRight": 9,
                        "paddingTop": 6,
                        "width": "100%",
                      },
                    ],
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <AnimatedComponent
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#5100FF",
                          "height": 39,
                          "justifyContent": "center",
                          "paddingBottom": 6,
                          "paddingLeft": 9,
                          "paddingRight": 9,
                          "paddingTop": 6,
                          "width": "100%",
                        },
                      ],
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#5100FF",
                        "height": 39,
                        "justifyContent": "center",
                        "opacity": 1,
                        "paddingBottom": 6,
                        "paddingLeft": 9,
                        "paddingRight": 9,
                        "paddingTop": 6,
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={false}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#5100FF",
                          "height": 39,
                          "justifyContent": "center",
                          "opacity": 1,
                          "paddingBottom": 6,
                          "paddingLeft": 9,
                          "paddingRight": 9,
                          "paddingTop": 6,
                          "width": "100%",
                        }
                      }
                    >
                      <Styled(Text)
                        color="#FFFFFF"
                        numberOfLines={1}
                      >
                        <StyledNativeComponent
                          color="#FFFFFF"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "displayName": "Styled(Text)",
                              "inlineStyle": InlineStyle {
                                "rules": Array [
                                  "
  color: ",
                                  [Function],
                                  ";
  font-family: 'AvenirNext-Bold';
  font-size: 14px;
  line-height: 19px;
  letter-spacing: 1px;
  text-align: center;
",
                                ],
                              },
                              "render": [Function],
                              "styledComponentId": "StyledNativeComponent",
                              "target": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          numberOfLines={1}
                        >
                          <Text
                            color="#FFFFFF"
                            numberOfLines={1}
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontFamily": "AvenirNext-Bold",
                                  "fontSize": 14,
                                  "letterSpacing": 1,
                                  "lineHeight": 19,
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            <Text
                              color="#FFFFFF"
                              numberOfLines={1}
                              style={
                                Array [
                                  Object {
                                    "color": "#FFFFFF",
                                    "fontFamily": "AvenirNext-Bold",
                                    "fontSize": 14,
                                    "letterSpacing": 1,
                                    "lineHeight": 19,
                                    "textAlign": "center",
                                  },
                                ]
                              }
                            >
                              SUPPRIMER
                            </Text>
                          </Text>
                        </StyledNativeComponent>
                      </Styled(Text)>
                      <PressabilityDebugView
                        color="cyan"
                      />
                    </View>
                  </View>
                </AnimatedComponent>
              </ForwardRef(AnimatedComponentWrapper)>
            </TouchableOpacity>
          </ForwardRef>
        </StyledNativeComponent>
      </Styled(Component)>
    </InputButton>
  </Component>
</ThemeContextProvider>
`;
